common/src/main/java/org/dash/wallet/common/transactions/TransactionUtils.kt

```diff
@@ -0,0 +1,121 @@
+import org.bitcoinj.core.TransactionBag
+import org.bitcoinj.script.ScriptException
+
+object TransactionUtils {
+    fun getWalletAddressOfReceived(tx: Transaction, bag: TransactionBag): Address? {
+        for (output in tx.outputs) {
+            try {
+                if (output.isMine(bag)) {
+                    return output.scriptPubKey.getToAddress(tx.params, true)
+                }
+            } catch (x: ScriptException) {
+                // swallow
+            }
+        }
+        return null
+    }
+
+    fun getFromAddressOfSent(tx: Transaction): List<Address> {
+        val result = mutableListOf<Address>()
+
+        for (input in tx.inputs) {
+            try {
+                val connectedTransaction = input.connectedTransaction
+                if (connectedTransaction != null) {
+                    val output = connectedTransaction.getOutput(input.outpoint.index)
+                    result.add(output.scriptPubKey.getToAddress(tx.params, true))
+                }
+            } catch (x: ScriptException) {
+                // swallow
+            }
+        }
+
+        return result
+    }
+
+    fun getToAddressOfReceived(tx: Transaction, bag: TransactionBag): List<Address> {
+        val result = mutableListOf<Address>()
+
+        for (output in tx.outputs) {
+            try {
+                if (output.isMine(bag)) {
+                    result.add(output.scriptPubKey.getToAddress(tx.params, true))
+                }
+            } catch (x: ScriptException) {
+                // swallow
+            }
+        }
+
+        return result
+    }
+
+    fun getToAddressOfSent(tx: Transaction, bag: TransactionBag): List<Address> {
+        val result = mutableListOf<Address>()
+
+        for (output in tx.outputs) {
+            try {
+                if (!output.isMine(bag)) {
+                    result.add(output.scriptPubKey.getToAddress(tx.params, true))
+                }
+            } catch (x: ScriptException) {
+                // swallow
+            }
+        }
+
+        return result
+    }
+
+    fun isEntirelySelf(tx: Transaction, bag: TransactionBag): Boolean {
+        for (input in tx.inputs) {
+            val connectedOutput = input.connectedOutput
+
+            if (connectedOutput == null || !connectedOutput.isMine(bag)) {
+                return false
+            }
+        }
+
+        for (output in tx.outputs) {
+            if (!output.isMine(bag)) {
+                return false
+            }
+        }
+
+        return true
+    }
+
+    val Transaction.allOutputAddresses: List<Address>
+        get() {
+            val result = mutableListOf<Address>()
+
+            outputs.forEach {
+                try {
+                    val script = it.scriptPubKey
+                    result.add(script.getToAddress(this.params, true))
+                } catch (x: ScriptException) {
+                    // swallow
+                }
+            }
+            return result
+        }
```

Comment:
Should this be moved up one level, meaning not part of TransactionUtils?  Is this used anywhere yet?

Reply:
Yes, it's used in the `TransactionResultViewBinder`. No problems, just needs a correct import.